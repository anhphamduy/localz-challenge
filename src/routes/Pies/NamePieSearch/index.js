import { Button, Input } from "antd";
import PropTypes from "prop-types";
import React from "react";

const NamePieSearch = ({
  dataIndex,
  setSelectedKeys,
  selectedKeys,
  onSearch
}) => {
  return (
    <div style={{ padding: 8 }}>
      <Input
        placeholder={`Search ${dataIndex}`}
        value={selectedKeys[0]}
        onChange={e => setSelectedKeys(e.target.value ? [e.target.value] : [])}
        onPressEnter={onSearch(selectedKeys)}
        style={{ width: 188, marginBottom: 8, display: "block" }}
      />
      <Button
        type="primary"
        onClick={onSearch(selectedKeys)}
        icon="search"
        size="small"
        style={{ marginRight: 8 }}
      >
        Search
      </Button>
      <Button
        onClick={() => {
          setSelectedKeys([]);
          onSearch("");
        }}
        size="small"
      >
        Reset
      </Button>
    </div>
  );
};

NamePieSearch.propTypes = {
  dataIndex: PropTypes.string.isRequired,
  setSelectedKeys: PropTypes.func.isRequired,
  selectedKeys: PropTypes.arrayOf(PropTypes.string).isRequired,
  onSearch: PropTypes.func.isRequired
};

export default NamePieSearch;
