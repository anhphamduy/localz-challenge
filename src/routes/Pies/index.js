import { message, Table } from "antd";
import React from "react";
import createUrl, { getPie } from "../../apis";
import NamePieSearch from "./NamePieSearch";

export default class Pies extends React.Component {
  state = {
    loading: true,
    pies: [],
    keyword: "",
    pagination: {
      showSizeChanger: true,
      current: 1,
      total: 1,
      pageSizeOptions: ["5", "10", "15"],
      defaultPageSize: 5
    },
    stores: []
  };

  componentDidMount() {
    Promise.resolve(this.fetch(this.state.pagination, undefined, {}));
  }

  onSearch = value => () => {
    this.setState(
      prevState => ({
        keyword: value,
        pagination: { ...prevState.pagination, current: 1 }
      }),
      () => {
        Promise.resolve(this.fetch(this.state.pagination, undefined, {}));
      }
    );
  };

  getColumnSearchProps = dataIndex => ({
    filterDropdown: ({ setSelectedKeys, selectedKeys }) => (
      <NamePieSearch
        selectedKeys={selectedKeys}
        setSelectedKeys={setSelectedKeys}
        onSearch={this.onSearch}
        dataIndex={dataIndex}
      />
    )
  });

  fetch = (pagination, _, sorter) => {
    this.setState({
      loading: true
    });
    return getPie(pagination.current, pagination.pageSize, this.state.keyword, {
      ...(sorter &&
        sorter.order && {
          _order: sorter.order === "ascend" ? "asc" : "desc",
          _sort: sorter.field
        })
    })
      .then(res => {
        // get the stores corresponding with the pies
        const savedStoreId = this.state.stores.map(store => store.id);
        const storeIds = Array.from(
          new Set(res.results.map(item => item.storeId))
        );

        // get a set of needed store ids
        const storePromises = storeIds
          .filter(storeId => !savedStoreId.includes(storeId))
          .map(storeId => {
            return fetch(createUrl("stores/" + storeId)).then(res =>
              res.json()
            );
          });

        return Promise.all(storePromises)
          .then(values => {
            this.setState(prevState => ({
              pies: [...res.results],
              loading: false,
              pagination: {
                ...pagination,
                total: Number(res.totalItems)
              },
              stores: [...prevState.stores, ...values]
            }));
          })
          .catch(err => {
            // logging error or something
            this.setState({ loading: false });
            message.error("Failed to fetch data");
          });
      })
      .catch(err => {
        // logging error or something
        this.setState({ loading: false });
        message.error("Failed to fetch data");
      });
  };

  tableColumns = [
    {
      title: "Pie Name",
      dataIndex: "displayName",
      render: name => name,

      sorter: true,
      ...this.getColumnSearchProps("displayName")
    },
    {
      title: "Quantity",
      dataIndex: "quantity",
      sorter: true,
      render: quantity => quantity
    },
    {
      title: "Price",
      dataIndex: "price",
      sorter: true,
      render: (_, record) => record.priceString
    },
    {
      title: "Pie Of The Day",
      dataIndex: "isPieOfTheDay",
      sorter: true,
      render: isPieOfTheDay => (isPieOfTheDay ? "Yes" : "No")
    },
    {
      title: "Store Rating",
      dataIndex: "rating",
      render: (_, record) =>
        this.state.stores.find(store => store.id === record.storeId).rating
    },
    {
      title: "Store Address",
      dataIndex: "address",
      render: (_, record) =>
        this.state.stores.find(store => store.id === record.storeId).address
    }
  ];

  render() {
    return (
      <React.Fragment>
        <Table
          columns={this.tableColumns}
          rowKey={record =>
            record.displayName +
            record.priceString +
            record.quantity +
            record.storeId
          }
          dataSource={this.state.pies}
          pagination={this.state.pagination}
          onChange={this.fetch}
          loading={this.state.loading}
        />
      </React.Fragment>
    );
  }
}
