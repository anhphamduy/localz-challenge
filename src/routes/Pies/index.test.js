import React from "react";
import { shallow } from "enzyme";
import Pies from "./index";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("<Pies />", async () => {
  const tree = shallow(<Pies />);
  const treeInstance = tree.instance()
  // test blank tree
  expect(treeInstance).toMatchSnapshot();

  // test tree data
  await Promise.resolve(treeInstance.fetch(
    {
      showSizeChanger: true,
      current: 1,
      total: 1,
      pageSizeOptions: ["5", "10", "15"],
      defaultPageSize: 5
    },
    undefined,
    {}
  ))
  expect(treeInstance).toMatchSnapshot();

  // test another tree data
  await Promise.resolve(treeInstance.fetch(
    {
      showSizeChanger: true,
      current: 2,
      total: 3,
      pageSizeOptions: ["5", "10", "15"],
      defaultPageSize: 10
    },
    undefined,
    {}
  ))
  expect(treeInstance).toMatchSnapshot();
});
