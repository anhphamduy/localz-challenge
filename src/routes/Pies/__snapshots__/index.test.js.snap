// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pies /> 1`] = `
Pies {
  "context": Object {},
  "fetch": [Function],
  "getColumnSearchProps": [Function],
  "onSearch": [Function],
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "keyword": "",
    "loading": true,
    "pagination": Object {
      "current": 1,
      "defaultPageSize": 5,
      "pageSizeOptions": Array [
        "5",
        "10",
        "15",
      ],
      "showSizeChanger": true,
      "total": 1,
    },
    "pies": Array [],
    "stores": Array [],
  },
  "tableColumns": Array [
    Object {
      "dataIndex": "displayName",
      "filterDropdown": [Function],
      "render": [Function],
      "sorter": true,
      "title": "Pie Name",
    },
    Object {
      "dataIndex": "quantity",
      "render": [Function],
      "sorter": true,
      "title": "Quantity",
    },
    Object {
      "dataIndex": "price",
      "render": [Function],
      "sorter": true,
      "title": "Price",
    },
    Object {
      "dataIndex": "isPieOfTheDay",
      "render": [Function],
      "sorter": true,
      "title": "Pie Of The Day",
    },
    Object {
      "dataIndex": "rating",
      "render": [Function],
      "title": "Store Rating",
    },
    Object {
      "dataIndex": "address",
      "render": [Function],
      "title": "Store Address",
    },
  ],
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <Pies />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": Object {
        "keyword": "",
        "loading": true,
        "pagination": Object {
          "current": 1,
          "defaultPageSize": 5,
          "pageSizeOptions": Array [
            "5",
            "10",
            "15",
          ],
          "showSizeChanger": true,
          "total": 1,
        },
        "pies": Array [],
        "stores": Array [],
      },
      "_rendered": <React.Fragment>
        <Table
          bordered={false}
          className=""
          columns={
            Array [
              Object {
                "dataIndex": "displayName",
                "filterDropdown": [Function],
                "render": [Function],
                "sorter": true,
                "title": "Pie Name",
              },
              Object {
                "dataIndex": "quantity",
                "render": [Function],
                "sorter": true,
                "title": "Quantity",
              },
              Object {
                "dataIndex": "price",
                "render": [Function],
                "sorter": true,
                "title": "Price",
              },
              Object {
                "dataIndex": "isPieOfTheDay",
                "render": [Function],
                "sorter": true,
                "title": "Pie Of The Day",
              },
              Object {
                "dataIndex": "rating",
                "render": [Function],
                "title": "Store Rating",
              },
              Object {
                "dataIndex": "address",
                "render": [Function],
                "title": "Store Address",
              },
            ]
          }
          dataSource={Array []}
          indentSize={20}
          loading={true}
          locale={Object {}}
          onChange={[Function]}
          pagination={
            Object {
              "current": 1,
              "defaultPageSize": 5,
              "pageSizeOptions": Array [
                "5",
                "10",
                "15",
              ],
              "showSizeChanger": true,
              "total": 1,
            }
          }
          rowKey={[Function]}
          showHeader={true}
          size="default"
          sortDirections={
            Array [
              "ascend",
              "descend",
            ]
          }
          useFixedHeader={false}
        />
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;

exports[`<Pies /> 2`] = `
Pies {
  "context": Object {},
  "fetch": [Function],
  "getColumnSearchProps": [Function],
  "onSearch": [Function],
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "keyword": "",
    "loading": false,
    "pagination": Object {
      "current": 1,
      "defaultPageSize": 5,
      "pageSizeOptions": Array [
        "5",
        "10",
        "15",
      ],
      "showSizeChanger": true,
      "total": 28,
    },
    "pies": Array [
      Object {
        "displayName": "Beef",
        "id": 1,
        "isPieOfTheDay": true,
        "price": 300,
        "priceString": "$3.00",
        "quantity": 5,
        "storeId": 1,
      },
      Object {
        "displayName": "Beef and curry",
        "id": 1,
        "price": 250,
        "priceString": "$2.50",
        "quantity": 2,
        "storeId": 1,
      },
      Object {
        "displayName": "Chicken",
        "id": 1,
        "price": 270,
        "priceString": "$2.70",
        "quantity": 0,
        "storeId": 1,
      },
      Object {
        "displayName": "Beef and bacon",
        "id": 1,
        "price": 290,
        "priceString": "$2.90",
        "quantity": 3,
        "storeId": 1,
      },
      Object {
        "displayName": "Kangaroo",
        "id": 1,
        "price": 300,
        "priceString": "$3.00",
        "quantity": 1,
        "storeId": 1,
      },
      Object {
        "displayName": "Beef",
        "id": 1,
        "price": 300,
        "priceString": "$3.00",
        "quantity": 0,
        "storeId": 2,
      },
      Object {
        "displayName": "Beef and curry",
        "id": 1,
        "isPieOfTheDay": true,
        "price": 250,
        "priceString": "$2.50",
        "quantity": 1,
        "storeId": 2,
      },
      Object {
        "displayName": "Chicken",
        "id": 1,
        "price": 350,
        "priceString": "$3.50",
        "quantity": 5,
        "storeId": 2,
      },
      Object {
        "displayName": "Beef and bacon",
        "id": 1,
        "price": 320,
        "priceString": "$3.20",
        "quantity": 2,
        "storeId": 2,
      },
      Object {
        "displayName": "Kangaroo",
        "id": 1,
        "price": 370,
        "priceString": "$3.70",
        "quantity": 2,
        "storeId": 2,
      },
    ],
    "stores": Array [
      Object {
        "address": "410 Collins Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.816981",
          "longitude": "144.9584893",
        },
        "displayName": "Eleven-7",
        "id": 1,
        "mobile": "0481233123",
        "postcode": "3000",
        "rating": 10,
        "state": "Victoria",
      },
      Object {
        "address": "366 Lonsdale Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.8128448",
          "longitude": "144.9604634",
        },
        "displayName": "Brian's Pies",
        "id": 2,
        "mobile": "0486162311",
        "postcode": "3000",
        "rating": 8,
        "state": "Victoria",
      },
    ],
  },
  "tableColumns": Array [
    Object {
      "dataIndex": "displayName",
      "filterDropdown": [Function],
      "render": [Function],
      "sorter": true,
      "title": "Pie Name",
    },
    Object {
      "dataIndex": "quantity",
      "render": [Function],
      "sorter": true,
      "title": "Quantity",
    },
    Object {
      "dataIndex": "price",
      "render": [Function],
      "sorter": true,
      "title": "Price",
    },
    Object {
      "dataIndex": "isPieOfTheDay",
      "render": [Function],
      "sorter": true,
      "title": "Pie Of The Day",
    },
    Object {
      "dataIndex": "rating",
      "render": [Function],
      "title": "Store Rating",
    },
    Object {
      "dataIndex": "address",
      "render": [Function],
      "title": "Store Address",
    },
  ],
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <Pies />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": Object {
        "keyword": "",
        "loading": false,
        "pagination": Object {
          "current": 1,
          "defaultPageSize": 5,
          "pageSizeOptions": Array [
            "5",
            "10",
            "15",
          ],
          "showSizeChanger": true,
          "total": 28,
        },
        "pies": Array [
          Object {
            "displayName": "Beef",
            "id": 1,
            "isPieOfTheDay": true,
            "price": 300,
            "priceString": "$3.00",
            "quantity": 5,
            "storeId": 1,
          },
          Object {
            "displayName": "Beef and curry",
            "id": 1,
            "price": 250,
            "priceString": "$2.50",
            "quantity": 2,
            "storeId": 1,
          },
          Object {
            "displayName": "Chicken",
            "id": 1,
            "price": 270,
            "priceString": "$2.70",
            "quantity": 0,
            "storeId": 1,
          },
          Object {
            "displayName": "Beef and bacon",
            "id": 1,
            "price": 290,
            "priceString": "$2.90",
            "quantity": 3,
            "storeId": 1,
          },
          Object {
            "displayName": "Kangaroo",
            "id": 1,
            "price": 300,
            "priceString": "$3.00",
            "quantity": 1,
            "storeId": 1,
          },
          Object {
            "displayName": "Beef",
            "id": 1,
            "price": 300,
            "priceString": "$3.00",
            "quantity": 0,
            "storeId": 2,
          },
          Object {
            "displayName": "Beef and curry",
            "id": 1,
            "isPieOfTheDay": true,
            "price": 250,
            "priceString": "$2.50",
            "quantity": 1,
            "storeId": 2,
          },
          Object {
            "displayName": "Chicken",
            "id": 1,
            "price": 350,
            "priceString": "$3.50",
            "quantity": 5,
            "storeId": 2,
          },
          Object {
            "displayName": "Beef and bacon",
            "id": 1,
            "price": 320,
            "priceString": "$3.20",
            "quantity": 2,
            "storeId": 2,
          },
          Object {
            "displayName": "Kangaroo",
            "id": 1,
            "price": 370,
            "priceString": "$3.70",
            "quantity": 2,
            "storeId": 2,
          },
        ],
        "stores": Array [
          Object {
            "address": "410 Collins Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.816981",
              "longitude": "144.9584893",
            },
            "displayName": "Eleven-7",
            "id": 1,
            "mobile": "0481233123",
            "postcode": "3000",
            "rating": 10,
            "state": "Victoria",
          },
          Object {
            "address": "366 Lonsdale Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.8128448",
              "longitude": "144.9604634",
            },
            "displayName": "Brian's Pies",
            "id": 2,
            "mobile": "0486162311",
            "postcode": "3000",
            "rating": 8,
            "state": "Victoria",
          },
        ],
      },
      "_rendered": <React.Fragment>
        <Table
          bordered={false}
          className=""
          columns={
            Array [
              Object {
                "dataIndex": "displayName",
                "filterDropdown": [Function],
                "render": [Function],
                "sorter": true,
                "title": "Pie Name",
              },
              Object {
                "dataIndex": "quantity",
                "render": [Function],
                "sorter": true,
                "title": "Quantity",
              },
              Object {
                "dataIndex": "price",
                "render": [Function],
                "sorter": true,
                "title": "Price",
              },
              Object {
                "dataIndex": "isPieOfTheDay",
                "render": [Function],
                "sorter": true,
                "title": "Pie Of The Day",
              },
              Object {
                "dataIndex": "rating",
                "render": [Function],
                "title": "Store Rating",
              },
              Object {
                "dataIndex": "address",
                "render": [Function],
                "title": "Store Address",
              },
            ]
          }
          dataSource={
            Array [
              Object {
                "displayName": "Beef",
                "id": 1,
                "isPieOfTheDay": true,
                "price": 300,
                "priceString": "$3.00",
                "quantity": 5,
                "storeId": 1,
              },
              Object {
                "displayName": "Beef and curry",
                "id": 1,
                "price": 250,
                "priceString": "$2.50",
                "quantity": 2,
                "storeId": 1,
              },
              Object {
                "displayName": "Chicken",
                "id": 1,
                "price": 270,
                "priceString": "$2.70",
                "quantity": 0,
                "storeId": 1,
              },
              Object {
                "displayName": "Beef and bacon",
                "id": 1,
                "price": 290,
                "priceString": "$2.90",
                "quantity": 3,
                "storeId": 1,
              },
              Object {
                "displayName": "Kangaroo",
                "id": 1,
                "price": 300,
                "priceString": "$3.00",
                "quantity": 1,
                "storeId": 1,
              },
              Object {
                "displayName": "Beef",
                "id": 1,
                "price": 300,
                "priceString": "$3.00",
                "quantity": 0,
                "storeId": 2,
              },
              Object {
                "displayName": "Beef and curry",
                "id": 1,
                "isPieOfTheDay": true,
                "price": 250,
                "priceString": "$2.50",
                "quantity": 1,
                "storeId": 2,
              },
              Object {
                "displayName": "Chicken",
                "id": 1,
                "price": 350,
                "priceString": "$3.50",
                "quantity": 5,
                "storeId": 2,
              },
              Object {
                "displayName": "Beef and bacon",
                "id": 1,
                "price": 320,
                "priceString": "$3.20",
                "quantity": 2,
                "storeId": 2,
              },
              Object {
                "displayName": "Kangaroo",
                "id": 1,
                "price": 370,
                "priceString": "$3.70",
                "quantity": 2,
                "storeId": 2,
              },
            ]
          }
          indentSize={20}
          loading={false}
          locale={Object {}}
          onChange={[Function]}
          pagination={
            Object {
              "current": 1,
              "defaultPageSize": 5,
              "pageSizeOptions": Array [
                "5",
                "10",
                "15",
              ],
              "showSizeChanger": true,
              "total": 28,
            }
          }
          rowKey={[Function]}
          showHeader={true}
          size="default"
          sortDirections={
            Array [
              "ascend",
              "descend",
            ]
          }
          useFixedHeader={false}
        />
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;

exports[`<Pies /> 3`] = `
Pies {
  "context": Object {},
  "fetch": [Function],
  "getColumnSearchProps": [Function],
  "onSearch": [Function],
  "props": Object {},
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "keyword": "",
    "loading": false,
    "pagination": Object {
      "current": 2,
      "defaultPageSize": 10,
      "pageSizeOptions": Array [
        "5",
        "10",
        "15",
      ],
      "showSizeChanger": true,
      "total": 28,
    },
    "pies": Array [
      Object {
        "displayName": "Pork",
        "id": 1,
        "price": 400,
        "priceString": "$4.00",
        "quantity": 2,
        "storeId": 2,
      },
      Object {
        "displayName": "Beef and curry",
        "id": 1,
        "price": 450,
        "priceString": "$4.50",
        "quantity": 1,
        "storeId": 3,
      },
      Object {
        "displayName": "Chicken",
        "id": 1,
        "price": 400,
        "priceString": "$4.00",
        "quantity": 5,
        "storeId": 3,
      },
      Object {
        "displayName": "Beef and bacon",
        "id": 1,
        "price": 300,
        "priceString": "$3.00",
        "quantity": 2,
        "storeId": 3,
      },
      Object {
        "displayName": "Beef and curry",
        "id": 1,
        "isPieOfTheDay": true,
        "price": 200,
        "priceString": "$2.00",
        "quantity": 1,
        "storeId": 3,
      },
      Object {
        "displayName": "Chicken",
        "id": 1,
        "price": 240,
        "priceString": "$2.40",
        "quantity": 5,
        "storeId": 3,
      },
      Object {
        "displayName": "Beef and bacon",
        "id": 1,
        "price": 280,
        "priceString": "$2.80",
        "quantity": 2,
        "storeId": 3,
      },
      Object {
        "displayName": "Beef and curry",
        "id": 1,
        "isPieOfTheDay": true,
        "price": 280,
        "priceString": "$2.80",
        "quantity": 1,
        "storeId": 4,
      },
      Object {
        "displayName": "Chicken",
        "id": 1,
        "price": 230,
        "priceString": "$2.30",
        "quantity": 5,
        "storeId": 4,
      },
      Object {
        "displayName": "Beef and bacon",
        "id": 1,
        "price": 230,
        "priceString": "$2.30",
        "quantity": 1,
        "storeId": 4,
      },
    ],
    "stores": Array [
      Object {
        "address": "410 Collins Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.816981",
          "longitude": "144.9584893",
        },
        "displayName": "Eleven-7",
        "id": 1,
        "mobile": "0481233123",
        "postcode": "3000",
        "rating": 10,
        "state": "Victoria",
      },
      Object {
        "address": "366 Lonsdale Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.8128448",
          "longitude": "144.9604634",
        },
        "displayName": "Brian's Pies",
        "id": 2,
        "mobile": "0486162311",
        "postcode": "3000",
        "rating": 8,
        "state": "Victoria",
      },
      Object {
        "address": "410 Collins Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.816981",
          "longitude": "144.9584893",
        },
        "displayName": "Eleven-7",
        "id": 1,
        "mobile": "0481233123",
        "postcode": "3000",
        "rating": 10,
        "state": "Victoria",
      },
      Object {
        "address": "366 Lonsdale Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.8128448",
          "longitude": "144.9604634",
        },
        "displayName": "Brian's Pies",
        "id": 2,
        "mobile": "0486162311",
        "postcode": "3000",
        "rating": 8,
        "state": "Victoria",
      },
      Object {
        "address": "345 Bourke Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.8130036",
          "longitude": "144.9646799",
        },
        "displayName": "Hungry Jake's",
        "id": 3,
        "mobile": "0412512333",
        "postcode": "3000",
        "rating": 9,
        "state": "Victoria",
      },
      Object {
        "address": "211 La Trobe Street",
        "city": "Melbourne",
        "coords": Object {
          "latitude": "-37.8100285",
          "longitude": "144.9628667",
        },
        "displayName": "Texcal",
        "id": 4,
        "mobile": "0498124512",
        "postcode": "3000",
        "rating": 3,
        "state": "Victoria",
      },
    ],
  },
  "tableColumns": Array [
    Object {
      "dataIndex": "displayName",
      "filterDropdown": [Function],
      "render": [Function],
      "sorter": true,
      "title": "Pie Name",
    },
    Object {
      "dataIndex": "quantity",
      "render": [Function],
      "sorter": true,
      "title": "Quantity",
    },
    Object {
      "dataIndex": "price",
      "render": [Function],
      "sorter": true,
      "title": "Price",
    },
    Object {
      "dataIndex": "isPieOfTheDay",
      "render": [Function],
      "sorter": true,
      "title": "Pie Of The Day",
    },
    Object {
      "dataIndex": "rating",
      "render": [Function],
      "title": "Store Rating",
    },
    Object {
      "dataIndex": "address",
      "render": [Function],
      "title": "Store Address",
    },
  ],
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_element": <Pies />,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_newState": Object {
        "keyword": "",
        "loading": false,
        "pagination": Object {
          "current": 2,
          "defaultPageSize": 10,
          "pageSizeOptions": Array [
            "5",
            "10",
            "15",
          ],
          "showSizeChanger": true,
          "total": 28,
        },
        "pies": Array [
          Object {
            "displayName": "Pork",
            "id": 1,
            "price": 400,
            "priceString": "$4.00",
            "quantity": 2,
            "storeId": 2,
          },
          Object {
            "displayName": "Beef and curry",
            "id": 1,
            "price": 450,
            "priceString": "$4.50",
            "quantity": 1,
            "storeId": 3,
          },
          Object {
            "displayName": "Chicken",
            "id": 1,
            "price": 400,
            "priceString": "$4.00",
            "quantity": 5,
            "storeId": 3,
          },
          Object {
            "displayName": "Beef and bacon",
            "id": 1,
            "price": 300,
            "priceString": "$3.00",
            "quantity": 2,
            "storeId": 3,
          },
          Object {
            "displayName": "Beef and curry",
            "id": 1,
            "isPieOfTheDay": true,
            "price": 200,
            "priceString": "$2.00",
            "quantity": 1,
            "storeId": 3,
          },
          Object {
            "displayName": "Chicken",
            "id": 1,
            "price": 240,
            "priceString": "$2.40",
            "quantity": 5,
            "storeId": 3,
          },
          Object {
            "displayName": "Beef and bacon",
            "id": 1,
            "price": 280,
            "priceString": "$2.80",
            "quantity": 2,
            "storeId": 3,
          },
          Object {
            "displayName": "Beef and curry",
            "id": 1,
            "isPieOfTheDay": true,
            "price": 280,
            "priceString": "$2.80",
            "quantity": 1,
            "storeId": 4,
          },
          Object {
            "displayName": "Chicken",
            "id": 1,
            "price": 230,
            "priceString": "$2.30",
            "quantity": 5,
            "storeId": 4,
          },
          Object {
            "displayName": "Beef and bacon",
            "id": 1,
            "price": 230,
            "priceString": "$2.30",
            "quantity": 1,
            "storeId": 4,
          },
        ],
        "stores": Array [
          Object {
            "address": "410 Collins Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.816981",
              "longitude": "144.9584893",
            },
            "displayName": "Eleven-7",
            "id": 1,
            "mobile": "0481233123",
            "postcode": "3000",
            "rating": 10,
            "state": "Victoria",
          },
          Object {
            "address": "366 Lonsdale Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.8128448",
              "longitude": "144.9604634",
            },
            "displayName": "Brian's Pies",
            "id": 2,
            "mobile": "0486162311",
            "postcode": "3000",
            "rating": 8,
            "state": "Victoria",
          },
          Object {
            "address": "410 Collins Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.816981",
              "longitude": "144.9584893",
            },
            "displayName": "Eleven-7",
            "id": 1,
            "mobile": "0481233123",
            "postcode": "3000",
            "rating": 10,
            "state": "Victoria",
          },
          Object {
            "address": "366 Lonsdale Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.8128448",
              "longitude": "144.9604634",
            },
            "displayName": "Brian's Pies",
            "id": 2,
            "mobile": "0486162311",
            "postcode": "3000",
            "rating": 8,
            "state": "Victoria",
          },
          Object {
            "address": "345 Bourke Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.8130036",
              "longitude": "144.9646799",
            },
            "displayName": "Hungry Jake's",
            "id": 3,
            "mobile": "0412512333",
            "postcode": "3000",
            "rating": 9,
            "state": "Victoria",
          },
          Object {
            "address": "211 La Trobe Street",
            "city": "Melbourne",
            "coords": Object {
              "latitude": "-37.8100285",
              "longitude": "144.9628667",
            },
            "displayName": "Texcal",
            "id": 4,
            "mobile": "0498124512",
            "postcode": "3000",
            "rating": 3,
            "state": "Victoria",
          },
        ],
      },
      "_rendered": <React.Fragment>
        <Table
          bordered={false}
          className=""
          columns={
            Array [
              Object {
                "dataIndex": "displayName",
                "filterDropdown": [Function],
                "render": [Function],
                "sorter": true,
                "title": "Pie Name",
              },
              Object {
                "dataIndex": "quantity",
                "render": [Function],
                "sorter": true,
                "title": "Quantity",
              },
              Object {
                "dataIndex": "price",
                "render": [Function],
                "sorter": true,
                "title": "Price",
              },
              Object {
                "dataIndex": "isPieOfTheDay",
                "render": [Function],
                "sorter": true,
                "title": "Pie Of The Day",
              },
              Object {
                "dataIndex": "rating",
                "render": [Function],
                "title": "Store Rating",
              },
              Object {
                "dataIndex": "address",
                "render": [Function],
                "title": "Store Address",
              },
            ]
          }
          dataSource={
            Array [
              Object {
                "displayName": "Pork",
                "id": 1,
                "price": 400,
                "priceString": "$4.00",
                "quantity": 2,
                "storeId": 2,
              },
              Object {
                "displayName": "Beef and curry",
                "id": 1,
                "price": 450,
                "priceString": "$4.50",
                "quantity": 1,
                "storeId": 3,
              },
              Object {
                "displayName": "Chicken",
                "id": 1,
                "price": 400,
                "priceString": "$4.00",
                "quantity": 5,
                "storeId": 3,
              },
              Object {
                "displayName": "Beef and bacon",
                "id": 1,
                "price": 300,
                "priceString": "$3.00",
                "quantity": 2,
                "storeId": 3,
              },
              Object {
                "displayName": "Beef and curry",
                "id": 1,
                "isPieOfTheDay": true,
                "price": 200,
                "priceString": "$2.00",
                "quantity": 1,
                "storeId": 3,
              },
              Object {
                "displayName": "Chicken",
                "id": 1,
                "price": 240,
                "priceString": "$2.40",
                "quantity": 5,
                "storeId": 3,
              },
              Object {
                "displayName": "Beef and bacon",
                "id": 1,
                "price": 280,
                "priceString": "$2.80",
                "quantity": 2,
                "storeId": 3,
              },
              Object {
                "displayName": "Beef and curry",
                "id": 1,
                "isPieOfTheDay": true,
                "price": 280,
                "priceString": "$2.80",
                "quantity": 1,
                "storeId": 4,
              },
              Object {
                "displayName": "Chicken",
                "id": 1,
                "price": 230,
                "priceString": "$2.30",
                "quantity": 5,
                "storeId": 4,
              },
              Object {
                "displayName": "Beef and bacon",
                "id": 1,
                "price": 230,
                "priceString": "$2.30",
                "quantity": 1,
                "storeId": 4,
              },
            ]
          }
          indentSize={20}
          loading={false}
          locale={Object {}}
          onChange={[Function]}
          pagination={
            Object {
              "current": 2,
              "defaultPageSize": 10,
              "pageSizeOptions": Array [
                "5",
                "10",
                "15",
              ],
              "showSizeChanger": true,
              "total": 28,
            }
          }
          rowKey={[Function]}
          showHeader={true}
          size="default"
          sortDirections={
            Array [
              "ascend",
              "descend",
            ]
          }
          useFixedHeader={false}
        />
      </React.Fragment>,
      "_rendering": false,
      "_updater": [Circular],
    },
  },
  Symbol(enzyme.__setState__): [Function],
}
`;
